File: python/2015/day2/main.py
totalArea = 0
totalLength = 0

with open("text.txt", "r") as file:
    for line in file:
        l,w,h = map(int, line.split('x'))
        areas = [l*w,l*h,w*h]
        totalArea +=  (2 * sum(areas)) + min(areas)

        totalLength += (2 * min(l+w,l+h,w+h)) + (l*w*h)

print(totalArea)
print(totalLength)

File: cpp/2015/day2/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <numeric>
#include <algorithm>

int main() {
    std::ifstream file {"text.txt"};
    std::string line;

    int i[3];
    int s[3];

    int total = 0;

    while (file >> i[0]) {
        file.ignore(1);
        file >> i[1];
        file.ignore(1);
        file >> i[2];

        s[0] = i[0] * i[1];
        s[1] = i[0] * i[2];
        s[2] = i[1] * i[2];

        total += *std::min_element(std::begin(s), std::end(s)) + (2 * (s[0]+s[1]+s[2])); 

    }

    std::cout << total << '\n';
}